### ISSUE TITLE
`@stylistic/operator-linebreak`: Comment nodes not properly handled

### ISSUE THREAD

**Issue by reteps** [eslint-stylistic/eslint-stylistic#870]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

snippet:

```ts
const showComments = true
const foo = (showComments)
? 'foo'
: // This has a lint error
  'bar'

const bar = (showComments)
  ? 'foo'
  : 'bar' // This does not have a lint error
```

config:

```ts
      '@stylistic/operator-linebreak': [
        'error',
        'before',
      ],
```

error:

>  Bad line breaking before and after '?'

I think that this isn't handling comment nodes correctly.

### Reproduction

https://stackblitz.com/edit/github-byuw45ap

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Hi! Thanks for reporting this!

This behavior is actually intentional - we have a specific test case for comments in this position:
[operator-linebreak.test.ts#L867](https://github.com/eslint-stylistic/eslint-stylistic/blob/ddd14c83cf77344a1775df745b1abca01543f5d6/packages/eslint-plugin/rules/operator-linebreak/operator-linebreak.test.ts#L867)

Personally, I'd recommend this format instead:

``` js
const bar = showComments
  ? 'foo'
  // some comments
  : 'bar'
```

Here is some historical context:
- https://github.com/eslint/eslint/commit/57bcafb0c5c21b864afc62559642349c0ecc9219
- https://github.com/eslint/eslint/commit/1beecece22471c3eefc822e29133b47d79c60dcd

I'm closing this for now, feel free to reopen if you have new insights.
